app/portainer/views/stacks/create/createStackController.js

```diff
@@ -133,12 +135,29 @@ function ($scope, $state, StackService, Authentication, Notifications, FormValid
     $scope.formValues.StackFileContent = cm.getValue();
   };
 
+  $scope.stackNameChange = function(name) {
+    $scope.stackNameAvailable = $scope.stackNames.indexOf(name) === -1;
+  };
+
   function initView() {
     var endpointMode = $scope.applicationState.endpoint.mode;
     $scope.state.StackType = 2;
     if (endpointMode.provider === 'DOCKER_SWARM_MODE' && endpointMode.role === 'MANAGER') {
       $scope.state.StackType = 1;
     }
+
+    StackService.stacks(
+      true,
+      endpointMode.provider === 'DOCKER_SWARM_MODE' && endpointMode.role === 'MANAGER',
```

Comment:
Stack name validation should be done against all endpoints for all type of providers, you can simply avoid the provider/role check here:

```suggestion
        true,
```